python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -652,3 +653,21 @@ def _isin_datetimelike(
 
     res = lhs._obtain_isin_result(rhs)
     return res
+
+
+def to_iso_calendar(self):
```

Comment:
(Possible nitpick): Is the input to this function a specific class or data structure? When I see self, I'm primed to think we're attached to a class, but this function is standalone. 

If this function gets a pandas Index or Series, could we ever hit `as_string_column` or `self.series` without throwing an error?

Reply:
We should probably throw an error in these cases. I should also probably not use self here either! Thanks for the catch. I'll change the code!

Reply:
It looks like this function is only used in our Series and Index classes, and doesn't accept any arguments beyond self. Is there a scenario in which we'd need to account for pandas-based inputs?

Reply:
It probably doen't! I'll take out that check :D